### DISCUSSION TITLE
Possibility to use Typescript enum types in results

### DISCUSSION THREAD

**Discussion by zaunermax** [prisma/prisma#2966] [Q&A] [answered]
Hello, first off I am really impressed by this lib. Especially the part with the partial selects but still having type safety!

My question is about enum types. I am a real Typescript enthusiast so I always work with as strict types as possible.

The question is if it is possible to make prisma results contain real Typescript enums in their results? I am talking about string enums, so not the bad enums :wink:. IMHO TS enums are more type safe than a string union.

**Comment by ryands17** [comment, marked as answer]
Hey @zaunermax :wave: 
Currently it's not possible to use directly as Prisma doesn't generate enums, so you would need to use enums in your TypeScript code as a workaround for now. Could you create a feature request regarding this [here](https://github.com/prisma/prisma/issues/new/choose) so that we can look into it further.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
